content/blog/2021-11-10-agent.md

```diff
@@ -0,0 +1,156 @@
+
+In this (lengthy) blog post, I would love to introduce a new operational mode of running Prometheus called "Agent". It is built directly into the Prometheus binary. The agent mode blocks some of the Prometheus features and optimizes the binary for scraping and remote writing to remote locations. While reducing the number of features on the solution sounds like a regression, I will explain why it is a game-changer for certain deployments in the CNCF ecosystem. I am super excited about this! (:
+
+## History of the Forwarding Use Case
+
+The core design of Prometheus has been unchanged for the whole project lifetime. Inspired by the [Google Borgmon](https://sre.google/sre-book/practical-alerting/#the-rise-of-borgmon), you can deploy it next to the application you want to monitor, tell Prometheus where they are and allow Prometheus to scrape the current values of their metrics every configured interval. Such a collection method, which is often referred to as the "pull model", is the core principle that allows Prometheus to be lightweight and ultra-reliable. Furthermore, it will enable applications and exporters to be dead simple as they need to provide a simple human-readable HTTP page with current metric values (in Open Metric format), without complex push infrastructure and non-trivial client libraries. Overall, in simple view, typical Prometheus monitoring deployment looks as presented below:
+
+![Prometheus high-level view](/assets/blog/2021-11-10/prom.png)
+
+This works great, and we have seen millions of successful deployments like this over the years that process dozens of millions of active series. Some of them for longer time retention, like two years so. All allow to query, alert, and record metrics useful for both cluster admins and developers.
+
+However, the CNCF world constantly grows and evolves. With the growth of managed Kubernetes solutions and clusters created on-demand within seconds, we are now finally able to treat "clusters" as "cattle", not as "pets" (in other words, we have more of those, how many cats you can have in your home? 🙃 ). In some cases, solutions do not even have the cluster notion anymore, e.g. [kcp](https://github.com/kcp-dev/kcp), [Google Cloud Run](https://cloud.google.com/run), [Fargate](https://aws.amazon.com/fargate/) and other serverless-like platforms.
```

Comment:
```suggestion
However, the cloud-native world is constantly growing and evolving. With the growth of managed Kubernetes solutions and clusters created on-demand within seconds, we are now finally able to treat clusters as "cattle", not as "pets" (in other words, we have more of those, how many cats you can have in your home? 🙃 ). In some cases, solutions do not even have the cluster notion anymore, e.g. [kcp](https://github.com/kcp-dev/kcp), [Google Cloud Run](https://cloud.google.com/run), [Fargate](https://aws.amazon.com/fargate/) and other serverless-like platforms.
```

"cattle" vs. "pets" is not mainly about having more of things, but treating things less like an individual, so the "too many cats in a home" comparison seems a bit odd to me?

You bring up Google Cloud Run etc., which as a reader would make me assume that with the "Agent" mode you are going to present me a solution now that will make Prometheus work for that use case, but it doesn't?